classes/i18n/Locale.php

```diff
@@ -389,6 +389,72 @@ public function getFormattedDisplayNames(array $filterByLocales = null, array $l
             ->toArray();
     }
 
+    /**
+     * Getting all translation strings needed in Vue.js UI
+     *
+     *
+     * @return  array Key are translation keys and values are translation strings
+     */
+    public function getUITranslationStrings(): array
```

Comment:
One thing it needs from Locale.php is the list of the registered paths, that could be exposed via public method.

Personally, I would like to move this new functionality to its own class, which would be used from the _i18n controller and from PKPTemplateManager. I always liked approach where individual features are implemented as 'services' in form of class or something to have clear public interface and abstract out implementation details on one place, rather than scatter them all over (controllers, models, ...).

But I am fine to follow whatever guidance I get as I am not familiar enough with existing code base and laravel abstractions to come up with something consistent with rest of the app.

Reply:
There is actually one more thing that would be good to get from Locale.php. And thats the cache key for getting locale bundler, because if I don't incorporate that to the cache key for translation strings, we still have the problem if someone edits po file - it won't get immediately picked up there.

So its actually quite closely connected with Locale.php..  so keeping this functionality there might be reasonable..

Reply:
Yep, I'm just sharing my opinion after skimming the code, I'll trust on your decisions 😁